### DISCUSSION TITLE
A way to copy the screen buffer

### DISCUSSION THREAD

**Discussion by eternalUnion** [mrcodetastic/ESP32-HUB75-MatrixPanel-DMA#175] [Q&A] [answered]
I'm looking for a way to copy the screen buffer (and pasting would be good as well) to a temp array (for an old project that used another matrix driver). Not really familiar to dma and I guess the buffer is stored in rows(?). Is there a way of doing that?

**Comment by mrcodetastic** [comment]
What format do you want the data in? The library stores the lowest level BCM signal in memory, that's it. You can't go backwards to say to an RGB888 format or anything. That information is lost already.

**Reply by eternalUnion** [reply]
could be in any format, as long as it can be pasted back to the buffer (I don't care about writing on the temp buffer I coppied currently)

**Reply by mrcodetastic** [reply]
The short answer is you can't do this with the library. I suggest implementing a pixel buffer prior to sending data to this library.

**Reply by eternalUnion** [reply]
Ok, I will overwrite the writePixel function, will update if some sort of result comes out of it.

**Comment by vortigont** [comment]
@eternalUnion,
if you like to experiment you can simply hack the lib and make 'frameStruct dma_buff' public, than you can access DMA buff structs directly and copy it back and forth via memcpy() anywhere you like, internally it's just an arrays if uint16_t's

**Comment by mrcodetastic** [comment, marked as answer]
There is a new example which implements an off-screen buffer at the risk of  using tonnes of memory:
https://github.com/mrfaptastic/ESP32-HUB75-MatrixPanel-I2S-DMA/discussions/225
